File: replace.mjs
import { promises as fs } from 'fs';

async function replaceInFile(filePath) {
  let content = await fs.readFile(filePath, 'utf8');
  content = content.replace(/د\.إ/g, 'ج.س');
  content = content.replace(/د\.ع/g, 'ج.س');
  content = content.replace(/بالدينار العراقي/g, 'بالجنيه السوداني');
  content = content.replace(/بالدينار/g, 'بالجنيه السوداني');
  await fs.writeFile(filePath, content, 'utf8');
  console.log(`Updated ${filePath}`);
}

async function run() {
  await replaceInFile('app/src/main/java/com/example/MainActivity.kt');
  await replaceInFile('app/src/main/java/com/example/ui/IncomeScreenView.kt');
  await replaceInFile('app/src/main/java/com/example/ui/ExpenseScreenView.kt');
}

run();

File: replace.py
import sys

with open('app/src/main/java/com/example/MainActivity.kt', 'r') as f:
    lines = f.readlines()

with open('patch_bottom_bar.kt', 'r') as f:
    patch_lines = f.readlines()

out_lines = lines[:108] + patch_lines + lines[322:]

with open('app/src/main/java/com/example/MainActivity.kt', 'w') as f:
    f.writelines(out_lines)

print("Done")
